tests/parser_tests.py

```diff
@@ -1176,6 +1176,44 @@ def test_iso8601_basic_format(self):
         with self.assertRaises(ParserError):
             self.parser.parse_iso("20180517T1055213Z")
 
+    def test_midnight_end_day(self):
+        self.assertEqual(
+            self.parser.parse_iso("2019-10-30T24:00:00"),
```

Comment:
`parse_iso` is only called when `arrow.get` is invoked without a format string (e.g. `arrow.get("2016-05-16")`). We should test this functionality on `self.parser.parse` as well to ensure that functionality like `arrow.get("2016-05-16T24:00:00", "YYYY-MM-DDTHH:mm:ss")` works as expected.

Reply:
@jadchaar At the end, both methods `parse_iso` and `parse` use `_build_datetime`, so no extra code changes were needed to support `parse`.
I added the test to ensure it will not break in the future.

Reply:
Ah yes you are correct! Missed that thanks to the myopic code collapsing in GitHub hah.